modules/acm/src/main/java/edu/gemini/epics/acm/CaObserveSenderImpl.java

```diff
@@ -728,7 +728,7 @@ public String signature() {
 
         @Override
         public CaObserveSenderImpl.ApplyState onCarValChange(final CarStateGeneric val) {
-            if ((carState == null || carState.isIdle()) && val.isBusy() && carClid > 0) {
+            if ((carState == null || carState.isIdle()) && val.isBusy() && carClid != null && carClid > 0) {
```

Comment:
The state event handlers are not called with null values, there is no need to check `val`.